src/App.jsx

```diff
@@ -3,21 +3,37 @@ import './App.css';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-function useSemiPersistentState() {
-  const [state, setState] = useState(() => {
+function App() {
+  const [todoList, setTodoList] = useState(() => {
     const savedState = localStorage.getItem('savedTodoList');
     return savedState ? JSON.parse(savedState) : [];
   });
 
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(state));
-  }, [state]);
+  const [isLoading, setIsLoading] = useState(true);
 
-  return [state, setState];
-}
+  useEffect(() => {
+    new Promise((resolve) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: []
+          }
+        });
+      }, 2000);
+    })
+      .then((result) => {
+        if (todoList.length === 0) {
+          setTodoList(result.data.todoList);
+        }
+        setIsLoading(false);
+      });
+  }, []);
 
-function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  useEffect(() => {
+    if (!isLoading && todoList.length > 0) {
```

Comment:
Adding the logic `todoList.length > 0` would cause issues with deleting todos when there's only one item left.

So I would suggest to remove it.